src/3_branch.js

```diff
@@ -301,8 +310,28 @@ Branch.prototype['init'] = wrap(callback_params.CALLBACK_ERR_DATA, function(done
 		done(err, data && utils.whiteListSessionData(data));
 	};
 
+	var allTheWaysToSayHidden =
+		[{ 'hidden': 'visibilitychange' },
+		{ 'mozHidden': 'mozvisibilitychange' },
+		{ 'msHidden': 'msvisibilitychange' },
+		{ 'webkitHidden': 'webkitvisibilitychange' }],
+
+	attachVisibilityEvent = function(data) {
+		for(var index = 0; index < allTheWaysToSayHidden.length; index++) {
+				var hidden = Object.keys(allTheWaysToSayHidden[index])[0],
+				event = allTheWaysToSayHidden[index][hidden];
+			if (typeof document[hidden] !== 'undefined') {
+				document['addEventListener'](event, function() {
+					if (!document[hidden]) { check_has_app(data, finishInit); }
```

Comment:
Yeah, probably a good point - let's not call finishInit again. Even if it doesn't have unintended side effects now, someone down the road may add something that could have unintended side effects thinking the function is only called when init is done.